packages/combobox/src/Combobox.tsx

```diff
@@ -0,0 +1,923 @@
+    switch (searchState) {
+      case 'loading': {
+        return (
+          <span className={menuMessage}>
+            <Icon
+              glyph="Refresh"
+              color={uiColors.blue.base}
+              className={loadingIconStyle}
+            />
+            {searchLoadingMessage}
+          </span>
+        );
+      }
+
+      case 'error': {
+        return (
+          <span className={menuMessage}>
+            <Icon glyph="Warning" color={uiColors.red.base} />
+            {searchErrorMessage}
+          </span>
+        );
+      }
+
+      case 'unset':
+      default: {
+        if (renderedOptions && renderedOptions.length > 0) {
+          return <ul className={menuList}>{renderedOptions}</ul>;
+        }
+
+        return <span className={menuMessage}>{searchEmptyMessage}</span>;
+      }
+    }
+  }, [
+    renderedOptions,
+    searchEmptyMessage,
+    searchErrorMessage,
+    searchLoadingMessage,
+    searchState,
+  ]);
+
+  const viewportSize = useViewportSize();
+
+  // Set the max height of the menu
+  const maxHeight = useMemo(() => {
+    // TODO - consolidate this hook with Select/ListMenu
+    const maxMenuHeight = 274;
+    const menuMargin = 8;
+
+    if (viewportSize && comboboxRef.current && menuRef.current) {
+      const {
+        top: triggerTop,
+        bottom: triggerBottom,
+      } = comboboxRef.current.getBoundingClientRect();
+
+      // Find out how much space is available above or below the trigger
+      const safeSpace = Math.max(
+        viewportSize.height - triggerBottom,
+        triggerTop,
+      );
+
+      // if there's more than enough space, set to maxMenuHeight
+      // otherwise fill the space available
+      return Math.min(maxMenuHeight, safeSpace - menuMargin);
+    }
+
+    return maxMenuHeight;
+  }, [viewportSize, comboboxRef, menuRef]);
+
+  // Scroll the menu when the focus changes
+  useEffect(() => {
+    // get the focused option
+  }, [focusedOption]);
+
+  /**
+   *
+   * Event Handlers
+   *
+   */
+  // Fired when the wrapper gains focus
+  const handleInputFocus = () => {
+    scrollToEnd();
+    openMenu();
+  };
+
+  // Fired onChange
+  const handleInputChange = ({
+    target: { value },
+  }: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(value);
+    // fire any filter function passed in
+    onFilter?.(value);
+  };
+
+  const handleClearButtonFocus = () => {
+    setFocusedOption(null);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    const isFocusInMenu = menuRef.current?.contains(document.activeElement);
+    const isFocusOnCombobox = comboboxRef.current?.contains(
+      document.activeElement,
+    );
+
+    const isFocusInComponent = isFocusOnCombobox || isFocusInMenu;
+
+    if (isFocusInComponent) {
+      // No support for modifiers yet
+      if (event.ctrlKey || event.shiftKey || event.altKey) {
+        return;
+      }
+
+      switch (event.keyCode) {
```

Comment:
We could also handle the delete button here. It would feel awesome if you could hit backspace when at the beginning of the input, and it deleted a selection.

Reply:
Or maybe it selects it instead like the left arrow key does 🤔 That might lead to less accidental deletions.

Reply:
At least three Options:
1) `Delete` key removes a chip (gmail)
2) `Delete` key selects the prev chip. Pressing again deletes it
3) `Delete` key removes a chip. and sets the input value to the chip's content (stackoverflow)

Reply:
I'd chat with Alven too about this, but I lean towards #2 to avoid accidental deletions while also supporting the backspace behavior.